src/error_chain.rs

```diff
@@ -65,20 +65,13 @@ macro_rules! impl_error_chain_processed {
         ///   - a backtrace, generated when the error is created.
         ///   - an error chain, used for the implementation of `Error::cause()`.
         #[derive(Debug)]
-        pub struct $error_name(
-            // The members must be `pub` for `links`.
-            /// The kind of the error.
-            pub $error_kind_name,
-            /// Contains the error chain and the backtrace.
-            #[doc(hidden)]
-            pub $crate::State,
-        );
+        pub struct $error_name(pub Box<($error_kind_name, $crate::State)>);
```

Comment:
Since we can't pattern match on the kind (`box` if nightly only), maybe we could remove `pub` and put everything into
```rust
struct ErrorInternals<K> {
  kind: K,
  backtrace: Backtrace,
  next: ...
}
```
I tried that at one time, but it was rejected because it prohibited the use of pattern matching. But if we can't use it anyway...

Reply:
At least in the usage I've seen I think it's quite important to match on `ErrorKind`, but is matching on the entire error itself that important?

Reply:
Yes : https://github.com/rust-lang-nursery/error-chain/blob/master/src/lib.rs#L348-L353
With the new method, you can't do that anymore. You have to match on Result, then on error.kind().

Reply:
Wrong link. Adding https://github.com/rust-lang-nursery/error-chain/pull/226

Reply:
That's true, yes.